classes/Page/Page.mjs

```diff
@@ -1,50 +1,50 @@
 import dbDriver from "../../database/driver.mjs"
 
 const databaseTiny = new dbDriver("tiny")
+const databaseMongo = new dbDriver("mongo")
 
 export default class Page {
-
     #tinyAction = 'create'
+
     #setRerumId() {
         if (this.#tinyAction === 'create') {
-            this.id = `${process.env.RERUMIDPREFIX}${id.split("/").pop()}`
+            this.id = `${process.env.RERUMIDPREFIX}${this.id.split("/").pop()}`
         }
         return this
     }
 
     /**
-     * Constructs a Page from the JSON Object in the Project `layers` Array reference. This 
-     * never creates a new Page, but rather wraps existing data in a Page object.
+     * Constructs a Page by wrapping existing data in a Page object.
      * Use the `build` method to create a new Page.
-     * @param {hexString} projectId For the project this layer belongs to
-     * @param {String} id The ID of the layer. This is the Layer stored in the Project.
-     * @param {String} label The label of the layer. This is the Layer stored in the Project.
-     * @param {String} target The uri of the targeted Canvas.
-     * @seeAlso {@link Page.build}
+     * @param {String} partOf The layer ID this page belongs to
+     * @param {Object} canvas An object with { id, label, target } properties
      */
-    constructor(layerId, { id, label, target }) {
+    constructor(partOf, canvas) {
+        if (!canvas || typeof canvas !== 'object') {
+            throw new Error("Page data is malformed.")
+        }
+        const { id, label, target } = canvas
         if (!id || !label || !target) {
             throw new Error("Page data is malformed.")
         }
-        Object.assign(this, { id, label, target, partOf: layerId })
-        if(this.id.startsWith(process.env.RERUMIDPREFIX)) {
+        Object.assign(this, { id, label, target, partOf })
+        if (this.id.startsWith(process.env.RERUMIDPREFIX)) {
             this.#tinyAction = 'update'
         }
         return this
     }
 
-    static build(layerId, canvas, prev, next, lines = []) {
+    static build(projectId, layerId, canvas, prev, next, lines = []) {
         if (!layerId) {
             throw new Error("Layer ID is required to create a Page instance.")
         }
         if (!canvas || !canvas.id) {
             throw new Error("Canvas with id is required to create a Page instance.")
         }
-
         const id = lines.length
             ? `${process.env.RERUMIDPREFIX}${databaseTiny.reserveId()}`
             : `${process.env.SERVERURL}layer/${layerId.split("/").pop()}/page/${databaseTiny.reserveId()}`
-        this.data = {
+        const pageDoc = {
```

Comment:
This is good and I think will conflict when you merge, but the other version is similar.